bundles/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/PulseaudioClient.java

```diff
@@ -369,76 +431,80 @@ public Optional<Integer> loadModuleSimpleProtocolTcpIfNeeded(AbstractAudioDevice
         return Optional.empty();
     }
 
+    public String getPAFormatString(AudioFormat format) {
+        assert AudioFormat.CODEC_PCM_SIGNED.equals(format.getCodec());
+        switch (Objects.requireNonNull(format.getBitDepth())) {
+            case 16:
+                return "s16" + (Objects.requireNonNull(format.isBigEndian()) ? "be" : "le");
+            case 24:
+                return "s24" + (Objects.requireNonNull(format.isBigEndian()) ? "be" : "le");
+            case 32:
+                return "s32" + (Objects.requireNonNull(format.isBigEndian()) ? "be" : "le");
+            default:
+                throw new IllegalArgumentException("Unsupported format : " + format.getBitDepth());
+        }
+    }
+
     /**
      * Find a simple protocol module corresponding to the given sink in argument
      * and returns the port it listens to
      *
      * @param item
      * @return
      */
-    private Optional<Integer> findSimpleProtocolTcpModule(AbstractAudioDeviceConfig item, @Nullable String format,
-            @Nullable BigDecimal rate, @Nullable BigDecimal channels) {
+    private Optional<SimpleProtocolTCPModule> findSimpleProtocolTcpModule(AbstractAudioDeviceConfig item,
+            @Nullable Integer id, @Nullable Integer port, @Nullable String format, @Nullable Long rate,
+            @Nullable Integer channels) {
         String itemType = getItemCommandName(item);
         if (itemType == null) {
             return Optional.empty();
         }
-        List<Module> modulesCopy = new ArrayList<>(modules);
-        var isSource = item instanceof Source;
-        return modulesCopy.stream() // iteration on modules
-                .filter(module -> MODULE_SIMPLE_PROTOCOL_TCP_NAME.equals(module.getPaName())) // filter on module name
```

Comment:
Did you remove ` .filter(module -> MODULE_SIMPLE_PROTOCOL_TCP_NAME.equals(module.getPaName()))  ` on purpose ?
(I'm not sure why it was here...)

Reply:
Yes, because now all of those are instances of SimpleProtocolTCPModule, and that is what I'm using to differentiate them from the other module instances.
```java
     ...
     modulesCopy.stream() //
        .filter(SimpleProtocolTCPModule.class::isInstance) //
        .map(SimpleProtocolTCPModule.class::cast) //
        ...
```